src/actions/config.js

```diff
@@ -18,23 +18,33 @@ export function fetchConfig() {
 }
 
 export function putConfig(config) {
-  return (dispatch) => {
-    let json;
-    try {
-      json = toJSON(config);
-    } catch (e) {
-      return dispatch(addNotification(getParserErrorMessage(), e.message, 'error'));
+  return (dispatch, getState) => {
+    const errors = validateConfig(config);
+    if (errors.length) {
+      return dispatch(validationError(errors));
```

Comment:
And also, we need to show the errors if any to users in the Configuration view. Check other views for guidance. Simply, you need to;

1. use `Errors` component to show them.
2. and clear errors when the component unmounts.